lib/howl/commands/app_commands.moon

```diff
@@ -639,16 +639,19 @@ file_search_hit_to_location = (match, search, display_as) ->
 
   loc
 
-do_search = (search, whole_word) ->
-  project = get_project!
+do_search = (directory, search, whole_word) ->
   file_search = howl.file_search
-  matches, searcher = file_search.search project.root, search, :whole_word
+  matches, searcher = file_search.search directory, search, :whole_word
   unless #matches > 0
     log.error "No matches found for '#{search}'"
     return matches
 
-  matches = file_search.sort matches, project.root, search, app.editor.current_context
-  display_as = project.config.file_search_hit_display
+  matches = file_search.sort matches, directory, search, app.editor.current_context
+
+  project = Project.for_file directory
+  conf = if project then project.config else config.for_file(directory)
```

Comment:
You don't need to determine the project config. The `config.for_file(directory)` is sufficient and will fallback to the project, or global config automatically.